async-aggregators/src/filter.py

```diff
@@ -0,0 +1,203 @@
+        title_box.setMinimumSize(200, 0)
+        title_box.setStyleSheet(
+            f"QFrame#{object_name + '_box'}"
+            "{ border: 1px solid 'black'; "
+            "border-radius: 6px; "
+            "background-color: 'white'; }"
+        )
+
+        title_centre = QHBoxLayout(title_box)
+        title_label = QLabel(name)
+        title_label.setStyleSheet("font-size: 22px")
+        title_centre.addWidget(QLabel())
+        title_centre.addWidget(title_label)
+        title_centre.addWidget(QLabel())
+
+        return title_box
+
+    @staticmethod
+    def style_slider(slider: QSlider, range_value: tuple, horizontal: bool) -> QFrame:
+        """
+        Style PyQt6 sliders for consistency
+
+        :param range_value:
+        :param slider:
+        :param slider, range_value, horizontal:
+        :return: QFrame slider
+        """
+        slider_frame = QFrame()
+        slider_frame.setObjectName("sliderframe")
+        slider_frame.setStyleSheet(
+            "QFrame#sliderframe { "
+            "border: 1px solid 'black';"
+            "border-radius: 6px;"
+            "background-color: 'white'; }"
+        )
+
+        slider_layout = (
+            QHBoxLayout(slider_frame) if horizontal else QVBoxLayout(slider_frame)
+        )
+        slider_layout.addWidget(QLabel(str(range_value[0])))
+        slider_layout.addWidget(slider)
+        slider_layout.addWidget(QLabel(str(range_value[-1])))
+
+        return slider_frame
+
+
+def apply_filter(filter_name: str, args: dict) -> QPixmap:
+    """
+    Apply a filter to an image
+
+    :param filter_name:
+    :param args:
+    :return: img
+    """
+    if filter_name == "Double Exposure":
+        args_for_filter = {}
+        for key, value in args.items():
+            if key == "Exposure":
+                args_for_filter["Exposure"] = value
+            if key == "image_to_edit":
+                args_for_filter["image_to_edit"] = value
+            if key == "second_image":
+                args_for_filter["second_image"] = value
+
+        return apply_double_exposure(
+            args_for_filter["image_to_edit"],
+            args_for_filter["second_image"],
+            args_for_filter["Exposure"],
+            args["image_label_w"],
+            args["image_label_h"],
+        )
+    if filter_name == "Ishihara":
+        args_for_filter = {}
+        for key, value in args.items():
+            if key == "A":
+                args_for_filter["A"] = value
+            if key == "B":
+                args_for_filter["B"] = value
+            if key == "image_to_edit":
+                args_for_filter["image_to_edit"] = value
+            if key == "image_label_w":
+                args_for_filter["image_label_w"] = value
+            if key == "image_label_h":
+                args_for_filter["image_label_h"] = value
+        return apply_unmask_reverse_ishihara(
+            args_for_filter
+        )
```

Comment:
I like splitting up long function calls onto multiple lines, but this can fit easily into one.
```suggestion
        return apply_unmask_reverse_ishihara(args_for_filter)
```